server/task_executor.go

```diff
@@ -155,8 +156,9 @@ func runMigration(ctx context.Context, l *zap.Logger, server *Server, task *api.
 	// If VCS based and schema path template is specified, then we will write back the latest schema file after migration.
 	writeBack := (vcsPushEvent != nil) && (repoRawOutter.SchemaPathTemplate != "")
 	// For tenant mode project, we will only write back latest schema file on the last task.
+	var project *api.Project
 	if writeBack && issue != nil {
-		project, err := server.composeProjectByID(ctx, task.Database.ProjectID)
+		project, err = server.composeProjectByID(ctx, task.Database.ProjectID)
```

Comment:
nit: let's move composeProjectByID before the line "if writeBack && issue != nil {" so that we can make sure it is not nil if used below.

Reply:
done